source/node.go

```diff
@@ -57,7 +57,11 @@ func NewNodeSource(ctx context.Context, kubeClient kubernetes.Interface, annotat
 	nodeInformer.Informer().AddEventHandler(
 		cache.ResourceEventHandlerFuncs{
 			AddFunc: func(obj interface{}) {
-				log.Debug("node added")
+				node, ok := obj.(*v1.Node)
+				if !ok {
+					log.Debug("node added")
```

Comment:
This construct is missing a return statement and should be an log.Error, otherwise we always going to have execution in `if !ok` block and next statement log.Debugf("node added: %s", node.ObjectMeta.Name)

Reply:
Hi @ivankatliarchuk,

The code does not return anything, that is why i don't throw an error.
Otherwise i have to check all the code paths if its handled correctly.

```
log.Debug("node added")
```

Reply:
It's an error case IF it's ever happens

```
log.Errorf("expected *v1.Node but got %T", obj)
```

Reply:
Is it always going to be of type `v1.Node`?
If that is true, then i think you are right.

Reply:
but consider a simple case 
- Type Mismatch: The value is a v1.Node (non-pointer) instead of *v1.Node (pointer).